src/ast.ts

```diff
@@ -326,20 +336,62 @@ export class AST {
   }
 }
 
+export type Pos = {
+  line: number;
+  ch: number;
+}
+
+export type NodeOptions = {
+  comment?: Comment;
+  "aria-label"?: string;
+}
+
 // Every node in the AST inherits from the `ASTNode` class, which is used to
 // house some common attributes.
-export class ASTNode {
-  from: any;
-  to: any;
-  type: any;
-  options: any;
-  id: string;
-  level: any;
+export abstract class ASTNode<Opt extends NodeOptions = NodeOptions, Props = {}> {
```

Comment:
This class gets used in a bunch of different ways that make it difficult to reason about what's going on. Some of the properties are clearly part of the public API, while others appear to be internal only, while still others are never actually used in this file but represent "sidecar" data that other internal components store on and retrieve from ASTNode instances.